lib/processors/manifestEnricher.js

```diff
@@ -0,0 +1,313 @@
+ * @returns {boolean}
+ */
+function isAutofillRequired(bundleConfig, appId) {
+	const {supportedLocales, bundleUrl} = bundleConfig;
+	let {bundleName} = bundleConfig;
+
+	if (supportedLocales) {
+		return false;
+	}
+
+	if (bundleUrl && bundleUrl.startsWith("/")) {
+		log.verbose("manifest.json: bundleUrl '" + bundleUrl +
+			"' contains an absolute path, no supportedLocales are generated");
+		return false;
+	}
+
+	if (bundleUrl && bundleUrl.startsWith("..")) {
+		log.verbose("manifest.json: bundleUrl '" + bundleUrl +
+			"' contains a relative path, no supportedLocales are generated");
+		return false;
+	}
+
+	if (bundleUrl && bundleUrl.startsWith("ui5://")) {
+		bundleName = getBundleNameByUrl(bundleUrl);
+	}
+
+	if (bundleName && !bundleName.startsWith(appId)) {
+		const logMsg = bundleUrl ? "bundleUrl '" + bundleUrl : "bundleName '" + bundleName;
+		log.verbose("manifest.json: " + logMsg +
+			"' contains a path which is not part of the project, no supportedLocales are generated");
+		return false;
+	}
+
+	return true;
+}
+
+function getType(manifest) {
+	return manifest["sap.app"].type;
+}
+
+function isFallbackLocalePartOfSupportedLocales(fallbackLocale, supportedLocales) {
+	return supportedLocales.includes(fallbackLocale);
+}
+
+function hasBundleTerminologies(bundleConfig) {
+	return bundleConfig?.terminologies && Object.keys(bundleConfig.terminologies).length > 0;
+}
+
+function getTerminologieBundles(bundleConfig, bundleConfigs) {
+	const terminologyBundles = [];
+	if (hasBundleTerminologies(bundleConfig)) {
+		Object.keys(bundleConfig.terminologies).forEach((key) => {
+			terminologyBundles.push(bundleConfig.terminologies[key]);
+		});
+	}
+	bundleConfigs.push(...terminologyBundles);
+}
+
+function getEnhanceWithBundles(bundleConfig, bundleConfigs) {
+	if (!bundleConfig.enhanceWith) {
+		return bundleConfigs;
+	}
+
+	bundleConfig.enhanceWith.forEach((config) => {
+		getTerminologieBundles(config, bundleConfigs);
+	});
+
+	bundleConfigs.push(...bundleConfig.enhanceWith);
+}
+
+function getSapAppBundle(manifest, bundleConfigs) {
+	const sapAppConfig = manifest["sap.app"];
+	let sapAppBundleConfig = sapAppConfig?.i18n;
+
+	if (getType(manifest) !== "library") {
+		if (hasManifestTemplates(manifest)) {
+			// sap/ui/core/Manifest defaults sap.app/i18n to "i18n/i18n.properties"
+			// when template is used in manifest.json
+			sapAppBundleConfig = sapAppBundleConfig || "i18n/i18n.properties";
+		} else {
+			sapAppBundleConfig = undefined;
+		}
+	}
+
+	if (!sapAppBundleConfig) {
+		return bundleConfigs;
+	}
+
+	if (typeof sapAppBundleConfig === "string" ) {
+		sapAppBundleConfig = {
+			bundleUrl: sapAppBundleConfig
+		};
+	} else {
+		getTerminologieBundles(sapAppBundleConfig, bundleConfigs);
+		getEnhanceWithBundles(sapAppBundleConfig, bundleConfigs);
+	}
+
+	sapAppConfig.i18n = sapAppBundleConfig;
+	bundleConfigs.push(sapAppBundleConfig);
+}
+
+function getSapUi5ModelBundles(manifest, bundleConfigs) {
```

Comment:
The name suggests that this should be a getter, but instead modifies `bundleConfigs`.

Reply:
I've updated the PR to rather call those methods `process...`, and they also modify the manifest, but only in case supportedLocales are added.